### PR TITLE
Remove trailing spaces from empty comments in `cabal init`

### LINKED ISSUE #7885: `cabal init` produces a trailing space in the generated .cabal file
This line

https://github.com/haskell/cabal/blob/master/cabal-install/src/Distribution/Client/Init/Format.hs#L294

results in a trailing space. I guess due to being an empty line prepended with "-- ".

### PR DESCRIPTION
**Pull request by ptkato**
Fixes #7885.

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

### CODE REVIEW

**Comment by ptkato** [comment]
It would be interesting to add this to the existing golden tests, however it seems that `tasty` ignores trailing spaces altogether.

**Review by Mikolaj — APPROVED** [review]
Thank you.

File: cabal-install/src/Distribution/Client/Init/Format.hs [hunk 1]
```diff
@@ -109,9 +109,11 @@ annNoComments :: FieldAnnotation
 annNoComments = FieldAnnotation False NoComment
 
 postProcessFieldLines :: FieldAnnotation -> [String] -> [String]
-postProcessFieldLines ann
-    | annCommentedOut ann = fmap ("-- " ++)
-    | otherwise = id
+postProcessFieldLines ann comments
+  | annCommentedOut ann = case () of
```
Comment (Mikolaj) [inline cabal-install/src/Distribution/Client/Init/Format.hs:113]: BTW, you can express this without the vacuous case over  `()`. e.g., with normal if or MultiWayIf (though hlint would complain there are only 2 cases for MultiWayIf).

**Comment by ptkato** [comment]
Turns out I was checking the wrong thing, and previously the change actually didn't change anything at all. Now it's all good, and tasty considers trailing spaces, yes.

**Review by Mikolaj — APPROVED** [review]
Huh, so tests are useful, after all. ;)

**Review by tchoutri — APPROVED** [review]

**Review by Kleidukos — APPROVED** [review]
Let's go

**Pull request merged**